service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/cmo/ReviewCMOController.java

```diff
@@ -0,0 +1,156 @@
+        Map<String, Object> data = caseDetails.getData();
+        CaseData caseData = mapper.convertValue(caseDetails.getData(), CaseData.class);
+
+        List<Element<CaseManagementOrder>> cmosReadyForApproval = reviewCMOService.getCMOsReadyForApproval(
+            caseData.getDraftUploadedCMOs());
+
+        data.remove("reviewCMODecision");
+
+        data.putAll(reviewCMOService.handlePageDisplayLogic(cmosReadyForApproval));
+
+        return AboutToStartOrSubmitCallbackResponse.builder()
+            .data(data)
+            .build();
+    }
+
+    @PostMapping("/mid-event")
+    public AboutToStartOrSubmitCallbackResponse handleMidEvent(@RequestBody CallbackRequest callbackRequest) {
+        CaseDetails caseDetails = callbackRequest.getCaseDetails();
+        Map<String, Object> data = caseDetails.getData();
+        CaseData caseData = mapper.convertValue(data, CaseData.class);
+
+        Element<CaseManagementOrder> selectedCMO = reviewCMOService.getSelectedCMO(caseData);
+
+        data.put("reviewCMODecision", ReviewDecision.builder()
+            .hearing(selectedCMO.getValue().getHearing())
+            .document(selectedCMO.getValue().getOrder())
+            .build());
+
+        List<Element<CaseManagementOrder>> cmosReadyForApproval = reviewCMOService.getCMOsReadyForApproval(
+            caseData.getDraftUploadedCMOs());
+
+        if (!(caseData.getCmoToReviewList() instanceof DynamicList)) {
+            // reconstruct dynamic list
+            data.put("cmoToReviewList", reviewCMOService.buildDynamicList(cmosReadyForApproval, selectedCMO.getId()));
+        }
+
+        return AboutToStartOrSubmitCallbackResponse.builder()
+            .data(data)
+            .build();
+    }
+
+    @PostMapping("/about-to-submit")
+    public AboutToStartOrSubmitCallbackResponse handleAboutToSubmit(@RequestBody CallbackRequest callbackRequest)
+        throws Exception {
+        CaseDetails caseDetails = callbackRequest.getCaseDetails();
+        Map<String, Object> data = caseDetails.getData();
+        CaseData caseData = mapper.convertValue(caseDetails.getData(), CaseData.class);
+
+        List<Element<CaseManagementOrder>> cmosReadyForApproval = reviewCMOService.getCMOsReadyForApproval(
+            caseData.getDraftUploadedCMOs());
+
+        if (!cmosReadyForApproval.isEmpty()) {
+            Element<CaseManagementOrder> cmo = reviewCMOService.getSelectedCMO(caseData);
+
+            if (!JUDGE_REQUESTED_CHANGES.equals(caseData.getReviewCMODecision().getDecision())) {
+                caseData.getDraftUploadedCMOs().remove(cmo);
+
+                Element<CaseManagementOrder> cmoToSeal = reviewCMOService.getCMOToSeal(caseData, cmo);
+
+                DocumentReference convertedDocument = documentConversionService.convertDocument(
+                    cmoToSeal.getValue().getOrder());
+                DocumentReference sealedDocument = documentSealingService.sealDocument(convertedDocument);
```

Comment:
I would try merge these 2 action maybe, performance is my concern as both transfer file, or at lease mark it with TODO